classes/class-ledyer-lco-gateway.php

```
@@ -339,6 +347,56 @@ public function process_payment_handler( $order_id ) {
 			return false;
 		}
 
+		/**
+		 * Process the payment for HPP/redirect checkout flow.
+		 *
+		 * @param object $order The WooCommerce order.
+		 *
+		 * @return array|string[]
+		 */
+		protected function hpp_redirect_handler( $order ) {
+
+			if ( empty( $order ) ) {
+				wc_add_notice( 'Failed to get order for HPP.', 'error' );
+				return array(
+					'result' => 'error',
+				);
+			}
+			$this->process_payment_handler( $order->get_id() );
+
+			$data = \Ledyer\Requests\Helpers\Woocommerce_Bridge::get_order_data( $order );
+			// Add confirmation URL to the order.
+			$ledyer_order = ledyer()->api->create_order_session( $data );
+			if ( is_wp_error( $ledyer_order ) ) {
+				wc_add_notice( 'Failed to create order session for HPP.', 'error' );
+				return array(
+					'result' => 'error',
+				);
+			}
+
+			// Create a HPP url.
+			$hpp_redirect = ledyer()->api->create_ledyer_hpp_url( $ledyer_order['sessionId'] );
+			if ( is_wp_error( $hpp_redirect ) ) {
+				wc_add_notice( 'Failed to create HPP session with ledyer.', 'error' );
```

`wc_add_notice` will only work if this is not a pay for order flow. The actual pay for order flow does not have notices, since it does not technically require a WC session. However i do believe you can return the errors in the return array under the messages key. But I'm not entirely sure on that. Could be good to test for example: 
```php
return array(
    'result' => 'error',
    'messages' => array( 'Failed to create HPP session with Ledyer' )
);
```